src/main/java/com/flightstats/hub/dao/aws/S3ConfigStrategy.java

```diff
@@ -82,8 +84,26 @@ static String getChannelTypedName(ChannelConfig config, String postfix) {
     private static BucketLifecycleConfiguration.Rule createRule(ChannelConfig config, String id) {
         return new BucketLifecycleConfiguration.Rule()
                 .withFilter(new LifecycleFilter(new LifecyclePrefixPredicate(id + "/")))
-                .withId(id)
+                .withId(BUCKET_LIFECYCLE_RULE_PREFIX.concat(id))
                 .withExpirationInDays((int) config.getTtlDays())
                 .withStatus(BucketLifecycleConfiguration.ENABLED);
     }
+
+    /**
+     * S3 Bucket lifecycle rules limit                         = 1000 per bucket
+     * Max S3 Bucket lifecycle rules Hub can create            = 990
+     * Max S3 Bucket lifecycle rules terraform code can create = 10
+     * The lifecycle rules created in hub only expires the objects based on TTL days configuration in hub channel.
+     * The expiration action in S3 adds delete marker to current object and does not delete it.
+     * The lifecycle rules set in terraform code cleans up all the delete markers and previous versions of the object in the S3 bucket.
+     **/
+    static List<BucketLifecycleConfiguration.Rule> getNonHubBucketLifecycleRules(BucketLifecycleConfiguration config) {
+        List<BucketLifecycleConfiguration.Rule> currentBucketLifecycleRules = config.getRules();
+        List<BucketLifecycleConfiguration.Rule> filteredRules =
+                currentBucketLifecycleRules
+                        .parallelStream()
+                        .filter(rule -> !rule.getId().contains(BUCKET_LIFECYCLE_RULE_PREFIX))
```

Comment:
I would use `.startsWith()` (I think?) instead of `.contains()` to not get false positives on rules that have "HUB_" somewhere inside them - if that's possible.

Reply:
that's logical.